code/aind_ccf_reg/plots.py

```diff
@@ -0,0 +1,202 @@
+    half_size_warped = np.array(warped.shape) // 2
+
+    if loc == 0:    
+        moving = moving.view()[half_size_moving[0], :, :]
+        fixed  = fixed.view()[half_size_fixed[0], :, :]
+        warped = warped.view()[half_size_warped[0], :, :]
+        y = 0.75
+    elif loc == 1:    
+        # moving = np.rot90(moving.view()[:,half_size[1], :], 3)
+        moving = moving.view()[:,half_size_moving[1], :]
+        fixed  = fixed.view()[:,half_size_fixed[1], :]
+        warped = warped.view()[:,half_size_warped[1], :]
+        y = 0.82
+    elif loc == 2:    
+        moving = np.rot90(np.fliplr(moving.view()[:, :, half_size_moving[2]]))
+        fixed  = np.rot90(np.fliplr(fixed.view()[:, :, half_size_fixed[2]]))
+        warped = np.rot90(np.fliplr(warped.view()[:, :, half_size_warped[2]]))
+        y = 0.82
+    else:
+        raise ValueError(
+            f"loc {loc} is not allowed. Allowed values are: 0, 1, 2")
+       
+    # combine deformed and fixed images to an RGB image
+    overlay = np.stack( (warped, fixed, warped), axis=2 )
+    diff = fixed - warped
+        
+    fontsize = 14
+    
+    fig, ax = plt.subplots(1, 5, figsize=(16, 6))
+    ax[0].imshow(moving, cmap='gray', vmin=vmin, vmax=vmax)
+    ax[1].imshow(fixed, cmap='gray', vmin=vmin, vmax=vmax)
+    ax[2].imshow(warped, cmap='gray', vmin=vmin, vmax=vmax)
+    ax[3].imshow(overlay)
+    ax[4].imshow(diff, cmap='gray', vmin=-(vmax), vmax=vmax)
+
+    ax[0].set_title("Moving", fontsize=fontsize)
+    ax[1].set_title("Fixed", fontsize=fontsize)
+    ax[2].set_title("Deformed", fontsize=fontsize)
+    ax[3].set_title("Deformed Overlay Fixed", fontsize=fontsize)
+    ax[4].set_title("Fixed - Deformed", fontsize=fontsize)
+    
+    fig.suptitle(title, size = 18, y=y)
+            
+    if figpath:
+        plt.savefig(figpath, bbox_inches = 'tight', pad_inches = 0.01)
+        plt.close()
+    else:
+        fig.show()   
+        
+        
+def plot_reg_before_after(moving, fixed, warped, figpath, title="", loc=0, vmin=0, vmax=1.5):
+    """
+    if moving and fixed images have same dimension, 
+    plot registration results: moving, fixed, deformed,
+    overlay and difference images before and after registration
+    
+    Parameters
+    ------------
+    moving: ANTsImage
+        Moving image
+    fixed: ANTsImage
+        Fixed image
+    warped: ANTsImage
+        Deformed image
+    figpath: PathLike
+        Path where the plot is going to be saved
+    title: str
+        Figure title
+    loc: int
+        Visualization direction
+    vmin, vmax: float
+        Set the color limits of the current image.
+    """
+    
+    if loc >= len(moving.shape):
+        raise ValueError(
+            f"loc {loc} is not allowed, should be less than are {len(moving.shape)}"
+    ) 
+    
+    half_size_moving = np.array(moving.shape) // 2
+    half_size_fixed = np.array(fixed.shape) // 2
+    half_size_warped = np.array(warped.shape) // 2
+
+    if loc == 0:    
```

Comment:
All of this code up to line 165 is the same as the function above. Consider using a helper function.